src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -32,21 +33,24 @@ internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToCo
 
                 obj = CreateObject(ref state.Current);
 
-                if (state.Current.PropertyIndex > 0)
+                if (argumentState.FoundPropertyCount > 0)
                 {
                     Utf8JsonReader tempReader;
 
-                    for (int i = 0; i < state.Current.PropertyIndex; i++)
+                    FoundProperty[]? properties = argumentState.FoundProperties;
+                    Debug.Assert(properties != null);
+
+                    for (int i = 0; i < argumentState.FoundPropertyCount; i++)
```

Comment:
I see - this is the key change (`FoundPropertyCount` vs `PropertyIndex`).

Reply:
Did this cause any bugs or is this just an optimization?

Reply:
The change brings about an optimization, but is now necessary because `PropertyIndex`, the number of `JsonPropertyInfo`s fetched from the class info cache (e.g. `MyPoco.Foo`, `"missing property"` which doesn't bind to a member and no extension property) could be greater than `FoundPropertyCount`, the number of properties we will actually set or send to an extension property (e.g. `MyPoco.Foo`).

fwi - `FoundPropertyCount` wasn't used previously in this non-async code path, only for async, mainly due to missing out on this optimization.